### PR TITLE
Parcelize all the SDK core models

### PR DESCRIPTION
**Pull request by roxanemace**
I have serialized all the models of the SDK core with [Parcelize](https://kotlinlang.org/docs/tutorials/android-plugin.html#parcelable-implementations-generator). It will allow passing the data between the Android components.

### CODE REVIEW

File: sdk-core/src/main/java/com/reach5/identity/sdk/core/models/Profile.kt [hunk 1]
```diff
@@ -28,11 +32,10 @@ data class Profile(
     // TODO better type for address
     val addresses: List<ProfileAddress>? = null,
     @SerializedName("custom_fields")
-    val customFields: Map<String, Any>? = null
-) {
+    val customFields: Map<String, @RawValue Any>? = null
```
Comment (axuan25) [inline sdk-core/src/main/java/com/reach5/identity/sdk/core/models/Profile.kt]: What does @RawValue do?

File: sdk-core/src/main/java/com/reach5/identity/sdk/core/models/RequestModels.kt [hunk 2]
```diff
@@ -53,11 +61,12 @@ data class RequestPasswordResetRequest (
     val redirectUrl: String? = null,
     @SerializedName("phone_number")
     val phoneNumber: String?
-)
+) : Parcelable
 
+@Parcelize
```
Comment (axuan25) [inline sdk-core/src/main/java/com/reach5/identity/sdk/core/models/RequestModels.kt]: Were these data classes parcelized before?

File: sdk-core/src/main/java/com/reach5/identity/sdk/core/models/User.kt [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
 import android.os.Parcelable
 import com.google.gson.annotations.SerializedName
+import kotlinx.android.parcel.Parcelize
 
+@Parcelize
 data class User(
-    /**
-     * Correspond au champ "sub" dans le JSON
-     */
+    // Correspond au champ "sub" dans le JSON
```
Comment (axuan25) [inline sdk-core/src/main/java/com/reach5/identity/sdk/core/models/User.kt]: Should we translate the comments into english for the SDK ?

File: sdk-core/src/main/java/com/reach5/identity/sdk/core/models/Profile.kt (hunk 1 above)
Comment (roxanemace) [reply sdk-core/src/main/java/com/reach5/identity/sdk/core/models/Profile.kt]: @rawvalue is an annotation required by @Parcelize to serialize `Any`
https://stackoverflow.com/questions/49606163/rawvalue-annotation-is-not-applicable-to-target-value-parameter
If you have a better idea, I am open to hear it 😄

File: sdk-core/src/main/java/com/reach5/identity/sdk/core/models/RequestModels.kt (hunk 2 above)
Comment (roxanemace) [reply sdk-core/src/main/java/com/reach5/identity/sdk/core/models/RequestModels.kt]: Nope there were not.

File: sdk-core/src/main/java/com/reach5/identity/sdk/core/models/User.kt (hunk 3 above)
Comment (roxanemace) [reply sdk-core/src/main/java/com/reach5/identity/sdk/core/models/User.kt]: That's a good point. I'll do it on my next PR.

**Comment by axuan25** [comment]
It would be nice to make chore changes such as the quotation marks in a separate chore PR, so as to avoid having to skip all the unnecessary diff. Minor nitpick :p

**Comment by roxanemace** [comment]
Isn't enough for this case to isolate the changes in different commits? https://github.com/ReachFive/identity-android-sdk/pull/19/commits

File: sdk-core/src/main/java/com/reach5/identity/sdk/core/models/RequestModels.kt (hunk 2 above)
Comment (axuan25) [reply sdk-core/src/main/java/com/reach5/identity/sdk/core/models/RequestModels.kt]: Why are we parcelizing them if they weren't ?

**Pull request merged**